### PR TITLE
📖 Consolidate content in the Cloud-Init and Deploy VM docs

### LINKED ISSUE #121: doc: Update docs/tutorials/deploy-vm/README.md
This ticket tracks updating [`./docs/tutorials/deploy-vm/README.md`](https://github.com/vmware-tanzu/vm-operator/tree/main/docs/tutorials/deploy-vm/README.md). Please refer to https://vm-operator.rtfd.io/en/latest/ref/proj/docs/#project-docs for guidelines on writing VM Operator project documentation.

### LINKED ISSUE #122: doc: Update docs/tutorials/deploy-vm/cloudinit.md
This ticket tracks updating [`./docs/tutorials/deploy-vm/cloudinit.md`](https://github.com/vmware-tanzu/vm-operator/tree/main/docs/tutorials/deploy-vm/cloudinit.md). Please refer to https://vm-operator.rtfd.io/en/latest/ref/proj/docs/#project-docs for guidelines on writing VM Operator project documentation.

### PR DESCRIPTION
**Pull request by sreyasn**
**What does this PR do, and why is it needed?**

  This change moves around information between docs that speaks of various ways to Deploy a VM using VM operator and moves the Cloud-init bootstrapping section to its own file.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #122, #121

----
:books: Documentation preview :books:: https://vm-operator--183.org.readthedocs.build/en/183/

### CODE REVIEW

**Review by zyiyi11 — APPROVED** [review]
minor change needed. LGTM. Thanks!

File: docs/tutorials/deploy-vm/cloudinit.md [hunk 1]
```diff
@@ -1,130 +1,20 @@
...
-            users:
-            - default
-        ```
-### Supported
-
-#### Cloud-Init
-
-[Cloud-Init](https://cloudinit.readthedocs.io/en/latest/) is widely recognized as the de facto method for bootstrapping modern VM instances on hyperscalers, including Tanzu with VM Operator. For example, the following resources may be used to deploy a VM and bootstrap its guest with Cloud-Init to:
+## Purpose
+As a DevOps user, you can bootstrap the guest inside a VM using the Cloud-Init to:
```
Comment (aruneshpa) [inline docs/tutorials/deploy-vm/cloudinit.md:10]: Not your change, but I don't think this is the exhaustive list of capabilities unlocked by cloud init user-data?  Maybe change the language to reflect that these are just an example of a few things that users can do.

File: docs/tutorials/deploy-vm/cloudinit.md [hunk 2]
```diff
@@ -1,130 +1,20 @@
 # Deploy a VM With Cloud-Init
 
-// TODO ([github.com/vmware-tanzu/vm-operator#122](https://github.com/vmware-tanzu/vm-operator/issues/122))
+This page reviews deploying a VM with the bootstrap method Cloud-Init.
 
-This page reviews the different components, workflows, and decisions related to deploying a VM with VM Operator:
+## Description
 
-## The VirtualMachine API
+[Cloud-Init](https://cloudinit.readthedocs.io/en/latest/) is widely recognized as the de facto method for bootstrapping modern VM instances on hyperscalers, including Tanzu with VM Operator.
```
Comment (aruneshpa) [inline docs/tutorials/deploy-vm/cloudinit.md:7]:
```suggestion
[Cloud-Init](https://cloudinit.readthedocs.io/en/latest/) is widely recognized as the de-facto method for bootstrapping modern VM instances on hyperscalers, including VM Service.
```

File: docs/tutorials/deploy-vm/cloudinit.md [hunk 3]
```diff
@@ -171,134 +61,13 @@ There are a number of methods that may be used to bootstrap a virtual machine's
...
-        <component name="Microsoft-Windows-Shell-Setup" processorArchitecture="amd64" publicKeyToken="31bf3856ad364e35" language="neutral" versionScope="nonSxS" xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
-          <OOBE>
-            <SkipMachineOOBE>true</SkipMachineOOBE>
-            <SkipUserOOBE>true</SkipUserOOBE>
-          </OOBE>
-        </component>
-      </settings>
-    </unattend>
-```
+The above example show a `VirtualMachine` custom resource using the CloudInit metadata transport provided by the `Secret` _my-vm-bootstrap-data_ .
```
Comment (aruneshpa) [inline docs/tutorials/deploy-vm/cloudinit.md:65]: Maybe reword it to something like?
```suggestion
The above example shows a `VirtualMachine` resource that specifies user-data using a `Secret` resource (`my-vm-bootstrap-data`), which will be used by `CloudInit` to bootstrap and customize the guest.
```

**Review by aruneshpa — APPROVED** [review]
LGTM with minor suggestions.

File: docs/tutorials/deploy-vm/README.md [hunk 4]
```diff
@@ -1,3 +1,248 @@
...
+      </settings>
+    </unattend>
+```
+
+For more information on Sysprep, please refer to Microsoft's [official documentation](https://learn.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/).
+
+#### vAppConfig
+
+The vAppConfig bootstrap method is useful for legacy, VM images that rely on bespoke, boot-time processes that leverage vAppConfig properties for customizing a guest.
+Refer to [Deploy a VM with vAppConfig](https://github.com/vmware-tanzu/vm-operator/blob/main/docs/tutorials/deploy-vm/vappconfig.md) for instructions to use this bootstrap method.
```
Comment (zyiyi11) [inline docs/tutorials/deploy-vm/README.md:236]: can you also change this link to `https://vm-operator.readthedocs.io/en/latest/tutorials/deploy-vm/vappconfig/`?

File: docs/tutorials/deploy-vm/cloudinit.md (hunk 2 above)
Comment (sreyasn) [reply docs/tutorials/deploy-vm/cloudinit.md]: making it "including VM Service on vSphere with Tanzu" since we're talking about hyperscalers.

**Review by aruneshpa — APPROVED** [review]
LGTM
